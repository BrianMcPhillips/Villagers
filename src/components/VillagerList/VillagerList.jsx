import React from 'react';
import PropTypes from 'prop-types';
import VillagerItem from '../VillagerItem/VillagerItem';
import styles from './VillagerList.module.css';

const VillagerList = ({ villagers }) => {
  const villagerElements = villagers.map(villager => {
    <li key={villager._id}>
      <VillagerItem name={villager.name} imageUrl={villager.imageUrl} />
    </li>
  });

  return (
    <ul className={styles.List}>
      {villagerElements}
    </ul>
  );
};

VillagerList.propTypes = {
  villagers: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    imageUrl: PropTypes.string.isRequired,
    _id: PropTypes.string.isRequired
  })).isRequired
};

export default VillagerList;
