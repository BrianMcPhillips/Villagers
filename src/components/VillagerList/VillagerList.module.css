.List {
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  list-style-type: none;
  align-items: end;
}