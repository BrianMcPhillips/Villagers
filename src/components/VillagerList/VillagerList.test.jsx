import React from 'react';
import { render, cleanup } from '@testing-library/react';
import VillagerList from './VillagerList';

describe('VillagerList component', () => {
  afterEach(() => cleanup());
  it('Renders a Villager List', () => {
    const { asFragment } = render(<VillagerList 
      villagers={[
        {
          name: 'Nicky Cage',
          imageUrl: 'https://placecage.com/200/200',
          _id: '28sjsh3jdjsh44444'
        },
        {
          name: 'Nicholas Cage',
          imageUrl: 'https://placecage.com/200/200',
          _id: 'jdjsuej23451jdj928'
        },
      ]}
    />);

    expect(asFragment()).toMatchSnapshot();
  });
});
