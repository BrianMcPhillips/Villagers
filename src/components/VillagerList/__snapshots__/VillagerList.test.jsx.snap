// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VillagerList component Renders a Villager List 1`] = `
<DocumentFragment>
  <ul
    class="List"
  />
</DocumentFragment>
`;
